tests/unit/time-series-test.js

```diff
@@ -359,3 +359,63 @@ test('Quarterly Filter', function(assert) {
       0, 'The filtered month is the start of a quarter');
   }
 });
+
+
+test('Minor Tick Filter', function(assert) {
+  assert.expect(13);
+  var startDate = new Date('2014-01-01'),
+    endDate = new Date('2016-01-01'),
+    component = this.subject(),
+    candidateLabels = d3.time.months(startDate.getTime(), endDate.getTime()),
+    expectedFiltered = [1391241600000, 1396335600000, 1401606000000, 1406876400000, 1412146800000, 1417420800000, 1422777600000, 1427871600000, 1433142000000, 1438412400000, 1443682800000, 1448956800000],
+    filteredCount = 0,
+    filteredMonths, len, item, i;
+
+  Ember.run(function() {
+    component.set('xAxis', {
+      selectAll: function() {
+        // return a list of dates out of order
+        var renderedLabels = d3.time.months(startDate.getTime(), endDate.getTime()),
+          currentIndex = renderedLabels.length,
+          temporaryValue, randomIndex;
+
+        // While there remain elements to shuffle...
+        while (0 !== currentIndex) {
+          // Pick a remaining element...
+          randomIndex = Math.floor(Math.random() * currentIndex);
```

Comment:
I agree that we should not use random even for the order. If the test fails, we should be able to reproduce with exact configuration when it fails.